glvis.cpp

```diff
@@ -1584,6 +1590,9 @@ int main (int argc, char *argv[])
          field_type = (use_soln) ? StreamState::FieldType::SCALAR
                       : StreamState::FieldType::MESH;
       }
+
+      GetMainThread();
```

Comment:
We can probably just move the call to `SDLMainLoop()` (which calls `GetMainThread`) to here, before the `single_session` construction, instead of after it.

Reply:
Since `SDLMainLoop()` blocks, moving it here is wrong.

Reply:
```suggestion
      // Make sure the singleton object returned by GetMainThread() is
      // initialized from the main thread.
      GetMainThread();
```